haystack/dataclasses/streaming_chunk.py

```diff
@@ -2,13 +2,17 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
+import warnings
 from dataclasses import dataclass, field
 from typing import Any, Awaitable, Callable, Dict, Literal, Optional, Union, overload
 
 from haystack.core.component import Component
 from haystack.dataclasses.chat_message import ToolCallResult
 from haystack.utils.asynchronous import is_callable_async_compatible
 
+# Type alias for standard finish_reason values following OpenAI's convention
+FinishReason = Literal["stop", "length", "tool_calls", "content_filter"]
```

Comment:
In addition to these let's add a Haystack specific one. Specifically we have a `tool_call_results` finish reason we set in `ToolInvoker` to indicate that the stream is finished. As in we send a final `StreamingChunk` that contains the `finish_reason` of `tool_call_results`.

Once we add this new finish reason let's make sure to set it using the new `finish_reason` field in `ToolInvoker`

Reply:
Right, right 👍